staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/composition.go

```diff
@@ -196,3 +197,39 @@ func (a *variableAccessor) Callback(_ *lazy.MapValue) ref.Val {
 	}
 	return v
 }
+
+func convertCelTypeToDeclType(celType *cel.Type) *apiservercel.DeclType {
```

Comment:
Is it possible to capture type params?

Reply:
Yes it is! Added support for type params.

However, note that the object is still Dyn so any access to its field remains Dyn.